lib/network/http_middlewares.go

```diff
@@ -125,19 +125,23 @@ func RateLimitMiddleware(logger logging.Logger, rule common.RateLimitRule) mux.M
 					middleware = middlewares[key]
 					middlewareCache.Add(ip, key)
 				} else if pip := net.ParseIP(ip); pip != nil {
+					var found bool
 					for inip, ipnet := range byCIDRs {
 						if !ipnet.Contains(pip) {
 							continue
 						}
 						key := middlewaresByIP[inip]
 						middleware = middlewares[key]
 						middlewareCache.Add(ip, key)
+						found = true
 						break
 					}
-				}
-			}
 
-			if middleware == nil {
+					if !found {
```

Comment:
@spikeekips `found` is always false here.

Reply:
@Charleslee522 Hmm, please see https://github.com/bosnet/sebak/pull/728/files/10dc54737c6ddecc74538f1619a170aeb3634473#diff-516f47ab88d984bfc549e69c298c2d5eR136

Reply:
@spikeekips Oh sorry, just ignore it :)